network/src/application/interface.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::application::{
+    db::PeerDb,
+    types::{PeerError, PeerInfo, PeerState},
+};
+use async_trait::async_trait;
+use diem_types::PeerId;
+use std::collections::{hash_map::Entry, HashMap};
+
+/// A generic `NetworkInterface` for applications to connect to networking
+#[async_trait]
+pub trait NetworkInterface {
+    type Sender;
+
+    /// Provides the PeerDb for other functions.  Not expected to be used externally.
+    fn db(&self) -> &PeerDb;
```

Comment:
So it actually looks like we will still maintain independent DBs but have a consistent implementation?

Reply:
It wasn't read only, because originally it was writing application data and scores into the `PeerDb`, but that had a poor reception, so I nixed it.

The purpose of this function was also because I was originally going to make `PeerDb` a trait, but rust didn't like what I was doing.

Reply:
> So it actually looks like we will still maintain independent DBs but have a consistent implementation?

@davidiw We have 1 DB, copying the `Arc` around of it.  This allows for the flexibility if we wanted to for some reason have multiple, but in reality they're all shared.